addons/mail/static/src/js/utils.js

```diff
@@ -37,10 +37,17 @@ function linkify(text, attrs) {
     attrs = _.map(attrs, function (value, key) {
         return key + '="' + _.escape(value) + '"';
     }).join(' ');
-    return text.replace(urlRegexp, function (url) {
+    var curIndex = 0;
+    var result = "";
+    var match;
+    while((match = urlRegexp.exec(text)) !== null) {
```

Comment:
exec is stateful (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec) so next call will set the internal index until not found and the internal index is set to 0

in the future this could be done more cleanly with the newer: [matchAll method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/matchAll#browser_compatibility) but there is no IE support it seems.

I need to have `curIndex` since when we have a match:

test http://example.com test

{match[0] => 'http://example.com', match.index => 5}

we need to get current index (0 at first) up to the index of the match (5) then adding the link.